src/main/java/io/dockstore/language/SnakemakeWorkflowPlugin.java

```diff
@@ -0,0 +1,172 @@
+/*
+ *    Copyright 2019 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.language;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import io.dockstore.common.DescriptorLanguage;
+import io.dockstore.common.VersionTypeValidation;
+import org.pf4j.Extension;
+import org.pf4j.Plugin;
+import org.pf4j.PluginWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author dyuen
+ */
+public class SnakemakeWorkflowPlugin extends Plugin {
+
+    public static final Logger LOG = LoggerFactory.getLogger(SnakemakeWorkflowPlugin.class);
+
+
+    /**
+     * Constructor to be used by plugin manager for plugin instantiation. Your plugins have to provide
+     * constructor with this exact signature to be successfully loaded by manager.
+     *
+     * @param wrapper
+     */
+    public SnakemakeWorkflowPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class SnakemakeWorkflowPluginImpl implements RecommendedLanguageInterface {
+
+        @Override
+        public String launchInstructions(String trsID) {
+            return null;
+        }
+
+
+        @Override
+        public VersionTypeValidation validateWorkflowSet(String initialPath, String contents, Map<String, FileMetadata> indexedFiles) {
+            VersionTypeValidation validation = new VersionTypeValidation(true, new HashMap<>());
+            // TODO hook up some real validation
+            return validation;
+        }
+
+        @Override
+        public VersionTypeValidation validateTestParameterSet(Map<String, FileMetadata> indexedFiles) {
+            return new VersionTypeValidation(true, new HashMap<>());
+        }
+
+
+        @Override
+        public DescriptorLanguage getDescriptorLanguage() {
+            return DescriptorLanguage.SMK;
+        }
+
+        @Override
+        public Pattern initialPathPattern() {
+            return Pattern.compile("\\.snakemake-workflow-catalog\\.yml");
```

Comment:
Why this particular file name?

Reply:
Thought we could piggy-back off descriptions, flags or other content from the Snakemake Catalog file https://github.com/snakemake/snakemake-workflow-catalog/tree/main

Reply:
Going to split the difference on this one, on one hand it is true that at minimum you only need a `Snakefile` 
On the other hand, according to the Snakemake workflow catalog readme, a `.snakemake-workflow-catalog.yml` is required for standardized reproducible use of Snakemake which is kind of our bread and butter.

Reply:
Cool, to me, it really does feel like `.snakemake-workflow-catalog.yml` is an important but non-essential metadata file (like `.dockstore.yml`) and not the primary descriptor.  There's probably Snakemake workflows out there that follow the prescribed directory structure, but don't have a `.snakemake-workflow-catalog.yml`, and this'll make it easier to tweak our system to support them.

Reply:
Yeah, in the catalog there are 330 "standardized workflows" and 3141 "other" workflows.